dotnet/src/Connectors/Connectors.OpenAI/Files/OpenAIFileService.cs

```diff
@@ -109,11 +109,22 @@ public async Task DeleteFileAsync(string id, CancellationToken cancellationToken
     /// <remarks>
     /// Files uploaded with <see cref="OpenAIFilePurpose.Assistants"/> do not support content retrieval.
     /// </remarks>
-    public BinaryContent GetFileContent(string id, CancellationToken cancellationToken = default)
+    public async Task<BinaryContent> GetFileContentAsync(string id, CancellationToken cancellationToken = default)
     {
         Verify.NotNull(id, nameof(id));
+        var (stream, mimetype) = await this.StreamGetRequestAsync($"{this._serviceUri}/{id}/content", cancellationToken).ConfigureAwait(false);
 
-        return new BinaryContent(() => this.StreamGetRequestAsync($"{this._serviceUri}/{id}/content", cancellationToken));
+        using (stream)
```

Comment:
Is there truely no way to avoid duplicating the stream to memory.  The file might be quite large.  This could create undue memory pressure for production services.

Reply:
I will be adding the `Stream` property to the `BinaryContent` in a following up PR for those scenarios where we don't want to allocate huge amounts of memory when using `BinaryContents`